GDJS/Runtime/events-tools/objecttools.ts

```diff
@@ -560,4 +560,53 @@ namespace gdjs {
       };
     }
   }
+
+  /**
+   * A container for objects lists that should last more than the current frame.
+   * It automatically removes objects that were destroyed from the objects lists.
+   */
+  export class LongLivedObjectsList {
+    private lists = new Map<string, Array<RuntimeObject>>();
+    private callbacks = new Map<RuntimeObject, () => void>();
+    private parent: LongLivedObjectsList | null = null;
+
+    static from(parent: LongLivedObjectsList): LongLivedObjectsList {
+      const newList = new LongLivedObjectsList();
+      newList.parent = parent;
+      return newList;
+    }
+
+    private getOrCreateList(o: string): RuntimeObject[] {
+      if (!this.lists.has(o)) this.lists.set(o, []);
+      return this.lists.get(o)!;
+    }
+
+    getObjects(name: string): RuntimeObject[] {
+      if (!this.lists.has(name) && this.parent)
+        return this.parent.getObjects(name);
+      return this.lists.get(name) || [];
+    }
+
+    addObject(o: gdjs.RuntimeObject): void {
```

Comment:
Could you avoid such abbreviation? When reading this, one needs to remember what is o (object? orientation? originalEvent?) on top of understanding the logic

Reply:
This is intensified as you used `o` as a `string` just above

Reply:
Whoops 😬 I was still prototyping when writing that code, forgot to clean it up. Thanks!